cpp/src/arrow/filesystem/localfs_test.cc

```diff
@@ -160,6 +187,32 @@ TYPED_TEST(TestLocalFS, FileSystemFromUriNoScheme) {
   this->TestFileSystemFromUri(this->local_path_);
 }
 
+TYPED_TEST(TestLocalFS, FileSystemFromUriFileBackslashes) {
+#ifdef _WIN32
+  this->TestFileSystemFromUri(ToBackslashes("file:" + this->local_path_));
+
+  // Variations
+  this->TestLocalUri("file:" + this->path_formatter_("C:foo\\bar"), "C:foo/bar");
```

Comment:
NB: `c:foo\bar` is a *relative* path, is this intended?

Reply:
Yes.